var/spack/repos/builtin/packages/py-urllib3/package.py

```diff
@@ -23,22 +25,30 @@ class PyUrllib3(PythonPackage):
     version("1.20", sha256="97ef2b6e2878d84c0126b9f4e608e37a951ca7848e4855a7f7f4437d5c34a72f")
     version("1.14", sha256="dd4fb13a4ce50b18338c7e4d665b21fd38632c5d4b1d9f1a1379276bd3c08d37")
 
-    variant("socks", default=False, description="SOCKS and HTTP proxy support")
+    variant("brotli", default=False, when="@1.25:", description="Add Brotli support")
     variant("secure", default=False, description="Add SSL/TLS support")
-    variant("brotli", default=False, description="Add Brotli support")
-
-    depends_on("python@2.7:2.8,3.4:", when="@:1.25", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@1.26.6", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.6:3", when="@1.26.12:", type=("build", "run"))
-
-    depends_on("py-setuptools", type="build")
-
-    depends_on("py-pyopenssl@0.14:", when="+secure")
-    depends_on("py-cryptography@1.3.4:", when="+secure")
-    depends_on("py-idna@2:", when="+secure")
-    depends_on("py-certifi", when="+secure")
-    depends_on("py-urllib3-secure-extra", when="+secure @1.26.12:")
-
-    depends_on("py-pysocks@1.5.6,1.5.8:1", when="+socks")
-
-    depends_on("py-brotlipy@0.6:", when="+brotli")
+    variant("socks", default=False, when="@1.15:", description="SOCKS and HTTP proxy support")
+
+    depends_on("py-hatchling@1.6:1", when="@2:", type="build")
+
+    with when("+brotli"):
+        depends_on("py-brotli@1.0.9:", when="@1.26: platform=linux", type=("build", "run"))
+        depends_on("py-brotli@1.0.9:", when="@1.26: platform=darwin", type=("build", "run"))
+        depends_on("py-brotli@1.0.9:", when="@1.26: platform=cray", type=("build", "run"))
+        depends_on("py-brotlipy@0.6:", when="@1.26: platform=windows", type=("build", "run"))
```

Comment:
Is this still true in 2.0.5? Can't remember where I got this from in the other PR, it's worth rechecking my math.

Reply:
For `@1.26.9:` it was
```
brotli>=1.0.9; (os_name != 'nt' or python_version >= '3') and platform_python_implementation == 'CPython' and extra == 'brotli'
brotlipy>=0.6.0; os_name == 'nt' and python_version < '3' and extra == 'brotli'
```
but for versions `@2:` its simplified to
```
"brotli>=1.0.9; platform_python_implementation == 'CPython'",
```
I overlooked the windows dependencies and thus thought the new ones are covered by this.
I have modified it now but was not sure how to handle windows. Till versions `@:1.26.8` `py-brotlipy` was a normal dependency, but with `@1.26.9:` they added `python_version<'3'` which is weird to me.
I cannot test windows, so either we leave it as is or I change it to too:
```
        depends_on("py-brotlipy@0.6:", when="@:1.26.8 platform=windows", type=("build", "run"))
        depends_on("py-brotlipy@0.6:", when="@1.25", type=("build", "run"))
```
Then there it no `py-brotlipy` dependency for version `@1.26.9:1.26`, which will likely not work.
What do you think?

Reply:
They added `and python_version<'3'` but they also added `or python_version >= '3'` for the other dep. So I think it should be `py-brotli@1.0.9:` when 1.26.9+.

Reply:
Your right, I overlooked the `or` and read it as `and`. Fixed now.